### DISCUSSION TITLE
Possible to use without Next.js or Express?

### DISCUSSION THREAD

**Discussion by mayank99** [trpc/trpc#2123] [General]
I like that the trpc client does not require react and can be used with any framework. But the trpc server seems very much designed for Next.js functions. I see there is an adapter for Express but I don't see any docs on creating a custom adapter.

My question is: how do I convert a trpc router into a serverless function that is compatible with the framework I'm currently using

For example, astro (and remix) supports API routes similar to next but the request/response syntax and types are different, and I would like to use trpc there. I'd imagine Sveltekit would have something similar, as well as any new SSR framework that comes out in the future.

I haven't looked at the ongoing v10 work so maybe this is already achievable with the rewrite?

**Comment by KATT** [comment]
- Third-party fetch adapter: https://github.com/valu-digital/npm-packages
- Standalone server: https://github.com/trpc/trpc/tree/main/examples/standalone-server
- Lambda adapter: https://trpc.io/docs/aws-lambda

**Comment by sachinraja** [comment]
Yes, this is achievable in v10. We decoupled from Node and added an official Fetch API adapter. This should allow you to use tRPC with Astro, SvelteKit, Remix, etc.

**Reply by mayank99** [reply]
Thank you, just found the PR #1852, amazing! Would be nice to reference [`fetch.md`](https://github.com/trpc/trpc/blob/next/www/docs/server/fetch.md) in [`v10-docs.md`](https://github.com/trpc/trpc/blob/next/.tmp/v10-docs.md)

**Reply by sachinraja** [reply]
Those docs are more of a "what's changed" instead of "what's been added" but we'll definitely reference it in any release notes/docs.

**Reply by moishinetzer** [reply]
This looks awesome! Any chance we could get a basic remix example whipped up?

**Reply by sachinraja** [reply]
Feel free to experiment on your own. Otherwise, I can try later when I have time but I don't have any experience with Remix.

**Reply by cj** [reply]
I have Astro working with v9; I'll post the repository next week :)

<img width="1638" alt="image" src="https://user-images.githubusercontent.com/1819/185759992-b7ea8075-6390-4d32-a0f0-405023d8b59c.png">

**Reply by sachinraja** [reply]
@cj Astro should also work with the [Fetch adapter](https://alpha.trpc.io/docs/fetch) for v10

**Reply by cj** [reply]
@sachinraja Yes - I am just using V9 right now as V10 hasn't been released and we use https://github.com/jlalmes/trpc-openapi. I'm going to package what I created into a trpc Astro adapter/client so others can use trpc with Astro easier.

**Reply by cj** [reply]
@sachinraja @mayank99  https://gist.github.com/cj/6bd9ffee73cf567285034f81c570a465

**Reply by mayank99** [reply]
amazing, thanks for creating this 👍 will try it out soon

**Reply by moishinetzer** [reply]
> Feel free to experiment on your own. Otherwise, I can try later when I have time but I don't have any experience with Remix.

Any chance you have some free time to make a quick example? It's quick 'n easy to start with the indie-stack repo by the remix team and convert that to use trpc I am confident it wouldn't take too long to do if you have the right trpc experience which I don't currently have. 

Kind thanks!
